### PR TITLE
Minor cleaning of Widget class, improve UX

### PR DESCRIPTION
**Pull request by devanych**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | #10

### CODE REVIEW

File: tests/WidgetTest.php [hunk 1]
```diff
@@ -117,13 +150,25 @@ public function testInjection(): void
         $this->assertInstanceOf(Injectable::class, $widget->getInjectable());
     }
 
-    public function testFactoryExceptionWidget(): void
+    public function testWidgetThrownExceptionForNotInitializeWidgetFactory(): void
```
Comment (vjik) [inline tests/WidgetTest.php:153]: May be we can way run this test first (configure PhpUnit) and don't user reflection ?

File: tests/WidgetTest.php [hunk 2]
```diff
@@ -117,13 +150,25 @@ public function testInjection(): void
...
+        $property->setAccessible(true);
+        $property->setValue($widgetFactoryReflection, null);
+        $property->setAccessible(false);
+
+        $this->expectException(WidgetFactoryInitializationException::class);
+        $this->expectExceptionMessage('Widget factory should be initialized with WidgetFactory::initialize() call.');
         TestWidget::widget()->id('w0')->render();
     }
+
+    public function testWidgetFactoryInitializationExceptionMessages(): void
```
Comment (vjik) [inline tests/WidgetTest.php:167]: Maybe merge this test with `testWidgetThrownExceptionForNotInitializeWidgetFactory` ?

Comment (devanych) [reply tests/WidgetTest.php:167]: In what way? In one case, the exception is thrown, and in the other, the custom methods of the exception class are checked.

File: tests/WidgetTest.php (hunk 1 above)
Comment (devanych) [reply tests/WidgetTest.php:153]: I don't see the point in redoing all the others because of one case.

File: tests/WidgetTest.php (hunk 2 above)
Comment (vjik) [reply tests/WidgetTest.php:167]: Ok

**Review by vjik — APPROVED** [review]

File: src/WidgetFactory.php [hunk 3]
```diff
@@ -10,6 +10,11 @@
 use Yiisoft\Factory\Exception\NotInstantiableException;
 use Yiisoft\Factory\FactoryInterface;
 
+/**
+ * WidgetFactory creates an instance of the widget based on the specified configuration
+ * {@see WidgetFactory::createWidget()}. Before creating a widget, need to initialize
+ * the WidgetFactory {@see WidgetFactory::initialize()}.
```
Comment (samdark) [inline src/WidgetFactory.php:16]:
```suggestion
 * WidgetFactory creates an instance of the widget based on the specified configuration
 * {@see WidgetFactory::createWidget()}. Before creating a widget, you need to initialize
 * the WidgetFactory with {@see WidgetFactory::initialize()}.
```

**Review by samdark — APPROVED** [review]

**Pull request merged**